Mini Cart V2 / Shopping Bag Footer section.

revanth0212 opened the pull request:
## Description

Added footer section for the new Mini Cart / Shopping Bag component.

**Should be merged only after #2509** 

## Related Issue
Closes [PWA-601](https://jira.corp.magento.com/browse/PWA-601)

### Verification Stakeholders
@jimbo 
@soumya-ashok 
@schensley 

### Verification Steps

1. run `yarn watch:venia`.
2. Add an item to the cart and click the new cart button.
3. We should be able to see the footer section.

## Screenshots / Screen Captures (if appropriate)

![image](https://user-images.githubusercontent.com/35203638/85621071-5211c780-b62a-11ea-8afb-00c835f4a7c3.png)

## Checklist
<!--- Go over all the following points, and make sure you've done anything necessary -->
- [ ] I have added tests to cover my changes, if necessary.
* **Should be merged only after #2509**

devops-pwa-codebuild commented:
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2511.pwa-venia.com : LH Performance Expected 0.85 Actual 0.53, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 79
https://pr-2511.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.36, LH Best Practices Expected 1 Actual 0.92
https://pr-2511.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.42, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

soumya-ashok commented:
@revanth0212 Feedback:

- Could we increase the padding above the checkout button and below "edit shopping bag"
- Font size on "edit shopping bag" looks very large. I know the mock has 16px - could we try dropping to 14px and see if that looks better?
- Button styling I assume will inherit from Dave's work? If not, the padding on L and R side of the label needs to be 24px - looks more than that now.
- We can remove the word "secure" from the CTA on the button and just have the lock icon with the word "checkout"

revanth0212 commented:
> @revanth0212 Feedback:
> 
> * Could we increase the padding above the checkout button and below "edit shopping bag"
> * Font size on "edit shopping bag" looks very large. I know the mock has 16px - could we try dropping to 14px and see if that looks better?
> * Button styling I assume will inherit from Dave's work? If not, the padding on L and R side of the label needs to be 24px - looks more than that now.
> * We can remove the word "secure" from the CTA on the button and just have the lock icon with the word "checkout"

@soumya-ashok I have addressed the review comments. Not sure of @davemacaulay's changes. I have updated the button to reflect similar to what you asked but it might not be what you expected. Ill have to check with the team about what is happening with buttons.

soumya-ashok commented:
@revanth0212 With the exception of the button, which I assume will be handled by @davemacaulay 's recent work, the other changes in this are UX approved. Thanks!

revanth0212 commented:
> @revanth0212 With the exception of the button, which I assume will be handled by @davemacaulay 's recent work, the other changes in this are UX approved. Thanks!

Dave mentioned his PR handles the button styling. If it does not work out after his PR is merged, we can revisit this PR.

revanth0212 commented:
Changed the CSS class names from underscore case to camel case.

tjwiebell commented on packages/peregrine/lib/talons/Header/useCartTrigger.js:
```diff
@@ -55,6 +55,7 @@ export const useCartTrigger = props => {
         miniCartIsOpen,
-        miniCartRef
+        miniCartRef,
+        setMiniCartIsOpen
```
Need to update snaps with this new prop

revanth0212 replied:
Maa badddd.. Forgot about that.

tjwiebell commented on packages/peregrine/lib/talons/MiniCart/__tests__/useItem.spec.js:
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Consider a test case for the unmount cleanup effect (useItem:L14)

revanth0212 replied:
I am still pondering if we even need that unmount logic. If the component is unmounting anyway, what's the point of setting the state? I must have forgotten to remove that piece of code

tjwiebell replied:
I agree with removal; local state doesn't matter if you're unmounting anyway.

revanth0212 replied:
Done removed.

dpatil-magento commented:
QA Approved.
